lib/services.js

```diff
@@ -9,37 +9,44 @@ import { HouseArrestService, HOUSE_ARREST_SERVICE_NAME } from './house-arrest';
 import PlistService from './plist-service';
 import semver from 'semver';
 
-async function startSyslogService (udid, socket) {
-  return new SyslogService(await startService(udid, SYSLOG_SERVICE_NAME, socket));
+async function startSyslogService (udid, opts = {}) {
+  return new SyslogService(await startService(udid, SYSLOG_SERVICE_NAME, opts.socket));
 }
 
 async function startSimulateLocationService (udid, socket) {
   return new SimulateLocationService(await startService(udid, SIMULATE_LOCATION_SERVICE_NAME, socket));
 }
 
-async function startWebInspectorService (udid, socket) {
-  const osVersion = await getOSVersion(udid, socket);
-  const semverVersion = semver.coerce(osVersion);
-  if (!semverVersion) {
-    throw new Error(`Couldn't create a semver version out of ${osVersion}`);
+async function startWebInspectorService (udid, opts = {}) {
+  let majorOsVersion = opts.majorOsVersion;
+  if (!majorOsVersion) {
+    const osVersion = await getOSVersion(udid, opts.socket);
+    const semverVersion = semver.coerce(osVersion);
+    if (!semverVersion) {
+      throw new Error(`Could not create a semver version out of '${osVersion}'`);
+    }
+    majorOsVersion = semverVersion.major;
   }
-  return new WebInspectorService(semverVersion.major, await startService(udid, WEB_INSPECTOR_SERVICE_NAME, socket));
+  const socketClient = opts.isSimulator
```

Comment:
I'm not really happy that we are passing an option called isSimulator to a package called `appium-ios-device` but I also don't have a better solution. One thing I thought about was moving the version comparison logic from `WebInspectorService` to here and have a boolean passed to `WebInspectorService` called `isPartialMessageSupported` instead of the major version. However, I don't have a strong opinion

Reply:
Or else it could be changed to be something like `shouldUseLockdown`, since that is all the sim check is doing. Then it is not tied explicitly to being a simulator.

Reply:
In terms of the version, in both sims and real devices the remote-debugger usually has the version and can pass it in.

Reply:
Maybe we can use that version then and lazily generate socket and version. If opts has those values we don't call getOSVersion and startService

Reply:
Do you mean something like
```js
async function startWebInspectorService (udid, opts = {}) {
  let majorOsVersion = opts.majorOsVersion;
  if (!majorOsVersion) {
    const osVersion = await getOSVersion(udid, opts.socket);
    const semverVersion = semver.coerce(osVersion);
    if (!semverVersion) {
      throw new Error(`Could not create a semver version out of '${osVersion}'`);
    }
    majorOsVersion = semverVersion.major;
  }
  const socketClient = opts.socket || await startService(udid, WEB_INSPECTOR_SERVICE_NAME);;
  return new WebInspectorService(majorOsVersion, socketClient);
}
```

Reply:
```javascript
async function startWebInspectorService (udid, opts = {}) {
  let osVersion = opts.osVersion || await getOSVersion(udid, opts.socket);
  const osVersion = await getOSVersion(udid, opts.socket);
  const semverVersion = semver.coerce(osVersion);
  if (!semverVersion) {
    throw new Error(`Could not create a semver version out of '${osVersion}'`);
  }
  majorOsVersion = semverVersion.major;
  const socketClient = opts.socket || await startService(udid, WEB_INSPECTOR_SERVICE_NAME);;
  return new WebInspectorService(majorOsVersion, socketClient);
}
```

Reply:
Sure.